hedera-node/hedera-app/src/main/java/com/hedera/node/app/workflows/handle/HandleWorkflow.java

```diff
@@ -201,15 +208,51 @@ private void handlePlatformTransaction(
             recordBuilder.status(ResponseCodeEnum.UNKNOWN);
         }
 
-        // TODO: handle long scheduled transactions
+        // store all records at once
+        final var recordListResult = recordListBuilder.build();
+        recordManager.endUserTransaction(recordListResult.recordStream());
+
+        // Verify if the transaction is a duplicate and add it to the cache
+        if (preHandleResult != null) { // this should always be true, checking just in case
```

Comment:
This is good to check, just in case! However, it raises the question: if we're worried this might happen and we are defending against it, then what do we do if it **IS** null? Do we log it? The transaction would be missing from the record cache, which would mean a replay of the transaction could happen, which would be VERY bad. Maybe it is better to die hard? Maybe we should add to the record cache BEFORE we commit things and write to the record stream so that if this happens, we do not end up committing any changes to state and we do not emit a record.

Reply:
Maybe intead of depending on the preHandleResult I can get the txBody and payer like we do in PreHandleWorkflowImpl? Something like:
```
final var txBytes = Bytes.wrap(platformTx.getContents());
final TransactionInfo txInfo;
try {
    txInfo = transactionChecker.parseAndCheck(txBytes);
} catch (PreCheckException preCheck) {
    // TODO: call recordFailedTransaction(e.responseCode(), recordBuilder, recordListBuilder); ?
}


final var txBody = txInfo.txBody();
final var payer = txId.accountID();
```
And I can remove the check altogether?